site/docs/types/named_structs.md

```diff
@@ -0,0 +1,79 @@
+# Named Structs
+
+A Named Struct is a special type construct that combines:
+* A Struct type
+* A list of names for the fields in the Struct, in depth-first search order
+
+The depth-first search order for names arises from the the ability to nest Structs within other types. All struct fields must be named, even nested fields.
+
+Named Structs are most commonly used to model the schema of Read relations.
+
+## Determining Names
+When producing/consuming names for a NamedStruct, some types requires special handling:
+
+### Struct
+A struct has names for each of its inner fields.
+
+For example, the following Struct
+```
+        a     b
+struct<i64, fp64>
+
+```
+has 2 names, one for each of its inner fields.
+
+### Nested Structs
```

Comment:
Basically this applies to any struct in any complex type (maps, lists, and structs).  Not sure if introducing that term makes the document simpler.

Reply:
Switched to Compound Types which is the language we use in the docs: https://substrait.io/types/type_classes/#compound-types